server/aws-lsp-codewhisperer/src/language-server/chat/chatSessionService.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    CodeWhispererStreaming,
+    GenerateAssistantResponseCommandInput,
+    GenerateAssistantResponseCommandOutput,
+} from '@amzn/codewhisperer-streaming'
+import { ConfiguredRetryStrategy } from '@aws-sdk/util-retry'
+import { CredentialsProvider } from '@aws/language-server-runtimes/server-interface'
+import { AbortController } from '@smithy/abort-controller'
+import { addProxyToClient } from 'aws-sdk-v3-proxy'
+import { getBearerTokenFromProvider } from '../utils'
+
+export class ChatSessionService {
+    public shareCodeWhispererContentWithAWS = false
+    readonly #codeWhispererRegion = 'us-east-1'
+    readonly #codeWhispererEndpoint = 'https://codewhisperer.us-east-1.amazonaws.com/'
+    #client: CodeWhispererStreaming
+    #abortController?: AbortController
+    #sessionId?: string
+
+    public get sessionId(): string | undefined {
+        return this.#sessionId
+    }
+
+    public set sessionId(value: string | undefined) {
+        this.#sessionId = value
+    }
+
+    constructor(credentialsProvider: CredentialsProvider) {
+        const steamingClient = new CodeWhispererStreaming({
+            region: this.#codeWhispererRegion,
+            endpoint: this.#codeWhispererEndpoint,
+            token: () => Promise.resolve({ token: getBearerTokenFromProvider(credentialsProvider) }),
+            retryStrategy: new ConfiguredRetryStrategy(0, (attempt: number) => 500 + attempt ** 10),
+        })
+
+        /**
+         * Streaming client use v3 sdk so there is no longer a global configuration for proxy
+         *
+         * addProxyToClient detects environment variables and determines whether or not to attach proxy
+         */
+
+        this.#client = addProxyToClient(steamingClient, { throwOnNoProxy: false })
```

Comment:
One note about using proxy: ideally servers developed in this repo should avoid using nodejs-specific libraries, to allow compiling and running the server as a  WebWorker in browser environments. `aws-sdk-v3-proxy` relies on NodeJS packages a lot, which would prevent the server to run in web.

This could be mitigated in future by introducing browser-compatible variants of this Service. 

In the past we have mitigated problem with running CodeWhisperer server behind proxy by exposing special "proxy" [variant of server](https://github.com/aws/language-servers/blob/main/server/aws-lsp-codewhisperer/src/language-server/proxy-server.ts).

But if that's currently not a goal to support browser, this should not be blocking from merging this PR

Reply:
Thanks for the call out! I think I will just move the `addProxyToClient` out to the proxy variant wrapper